packages/core2/src/managers/_RepositoryManager.ts

```diff
@@ -1,6 +1,255 @@
+	domain: string;
+	password: string;
+};
+
+type SliceMachineManagerPushDocumentsArgs = {
+	domain: string;
+	signature: string;
+	documents: Record<string, unknown>; // TODO: Type unknown if possible(?)
+};
+
 export class RepositoryManager extends BaseManager {
 	// TODO: Add methods for repository-specific actions. E.g. creating a
 	// new repository.
+
+	async readAll(): Promise<PrismicRepository[]> {
+		const url = new URL("/repositories", APIEndpoints.PrismicUser);
+		const res = await this._fetch({ url });
+		const json = await res.json();
+
+		if (res.ok) {
+			const { value: repositories, error } = decode(
+				t.array(PrismicRepository),
+				json,
+			);
+
+			if (error) {
+				throw new Error(
+					`Failed to decode repositories: ${error.errors.join(", ")}`,
+				);
+			}
+
+			return repositories;
+		} else {
+			throw new Error(`Failed to read repositories`, { cause: json });
+		}
+	}
+
+	// Should this be in manager? It's one of the few sync method
+	hasWriteAccess(repository: PrismicRepository): boolean {
+		switch (repository.role) {
+			case PrismicRepositoryRoles.SuperUser:
+			case PrismicRepositoryRoles.Owner:
+			case PrismicRepositoryRoles.Administrator:
+				return true;
+
+			default:
+				return false;
+		}
+	}
+
+	async checkExists(
+		args: SliceMachineManagerExistsRepositoryArgs,
+	): Promise<boolean> {
+		const url = new URL(
+			`/app/dashboard/repositories/${args.domain}/exists`,
+			APIEndpoints.PrismicWroom,
+		);
+		const res = await this._fetch({ url });
+		const text = await res.text();
+
+		if (res.ok) {
+			return text === "false"; // Endpoint returns `false` when repository exists
+		} else {
+			throw new Error(
+				`Failed to check repository existence for domain \`${args.domain}\``,
+				{ cause: text },
+			);
+		}
+	}
+
+	async create(args: SliceMachineManagerCreateRepositoryArgs): Promise<void> {
+		const url = new URL(
+			"/authentication/newrepository?app=slicemachine",
+			APIEndpoints.PrismicWroom,
+		);
+
+		const body = {
+			...DEFAULT_REPOSITORY_SETTINGS,
+			domain: args.domain,
+			framework: args.framework, // This property appears to be optional for the API
+		};
+
+		const res = await this._fetch({
+			url,
+			method: "POST",
+			body,
+			userAgent: RepositoryUserAgents.SliceMachine, // Custom User Agent is required
+		});
+		const text = await res.text();
+
+		// Endpoint returns repository name on success, which must be more than 4 characters and less than 30
+		// if (!res.ok) {
+		if (!res.ok || text.length < 4 || text.length > 30) {
+			throw new Error(`Failed to create repository \`${args.domain}\``, {
+				cause: res,
+			});
+		}
+	}
+
+	async delete(args: SliceMachineManagerDeleteRepositoryArgs): Promise<void> {
+		const cookies = await this.user.getAuthenticationCookies();
+
+		const url = new URL(
+			`/app/settings/delete?_=${cookies["X_XSRF"]}`, // TODO: Maybe we want to throw early if the token is no available, or get the token another way
```

Comment:
If we update the previously commented URL to use `url.searchParams`, we should make that change again here.